libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/LocationFpsDelegate.java

```diff
@@ -0,0 +1,92 @@
+package com.mapbox.services.android.navigation.ui.v5.map;
+
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxsdk.location.LocationComponent;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+
+import timber.log.Timber;
+
+class LocationFpsDelegate implements MapboxMap.OnCameraIdleListener {
+
+  private static final int ZOOM_LEVEL_FIVE = 5;
+  private static final int ZOOM_LEVEL_TEN = 10;
+  private static final int ZOOM_LEVEL_FOURTEEN = 14;
+  private static final int ZOOM_LEVEL_SIXTEEN = 16;
+  private static final int ZOOM_LEVEL_EIGHTEEN = 18;
+  private static final int MAX_ANIMATION_FPS_THREE = 3;
+  private static final int MAX_ANIMATION_FPS_FIVE = 5;
+  private static final int MAX_ANIMATION_FPS_FIFTEEN = 15;
+  private static final int MAX_ANIMATION_FPS_TWENTY = 20;
+  private static final int MAX_ANIMATION_FPS_THIRTY = 30;
+  private static final int MAX_ANIMATION_FPS = Integer.MAX_VALUE;
+  private final MapboxMap mapboxMap;
+  private final LocationComponent locationComponent;
+  private int currentFps = MAX_ANIMATION_FPS;
+  private boolean isEnabled = true;
+
+  LocationFpsDelegate(@NonNull MapboxMap mapboxMap, @NonNull LocationComponent locationComponent) {
+    this.mapboxMap = mapboxMap;
+    this.locationComponent = locationComponent;
+    mapboxMap.addOnCameraIdleListener(this);
+  }
+
+  @Override
+  public void onCameraIdle() {
+    if (!isEnabled) {
+      return;
+    }
+    updateMaxFps();
+  }
+
+  void onStart() {
+    mapboxMap.addOnCameraIdleListener(this);
+  }
+
+  void onStop() {
+    mapboxMap.removeOnCameraIdleListener(this);
+  }
+
+  void updateEnabled(boolean isEnabled) {
+    this.isEnabled = isEnabled;
+    resetMaxFps(!isEnabled);
+  }
+
+  boolean isEnabled() {
+    return isEnabled;
+  }
+
+  private void updateMaxFps() {
+    double zoom = mapboxMap.getCameraPosition().zoom;
+    int maxAnimationFps = buildFpsFrom(zoom);
+    if (currentFps != maxAnimationFps) {
+      Timber.d("Updating fps %s", maxAnimationFps);
+      locationComponent.setMaxAnimationFps(maxAnimationFps);
+      currentFps = maxAnimationFps;
+    }
+  }
+
+  private int buildFpsFrom(double zoom) {
+    int maxAnimationFps;
+    if (zoom < ZOOM_LEVEL_FIVE) {
+      maxAnimationFps = MAX_ANIMATION_FPS_THREE;
+    } else if (zoom < ZOOM_LEVEL_TEN) {
+      maxAnimationFps = MAX_ANIMATION_FPS_FIVE;
+    } else if (zoom < ZOOM_LEVEL_FOURTEEN) {
+      maxAnimationFps = MAX_ANIMATION_FPS_FIFTEEN;
+    } else if (zoom < ZOOM_LEVEL_SIXTEEN) {
+      maxAnimationFps = MAX_ANIMATION_FPS_TWENTY;
+    } else if (zoom < ZOOM_LEVEL_EIGHTEEN) {
+      maxAnimationFps = MAX_ANIMATION_FPS_THIRTY;
+    } else {
+      maxAnimationFps = MAX_ANIMATION_FPS;
+    }
+    return maxAnimationFps;
+  }
+
+  private void resetMaxFps(boolean shouldReset) {
+    if (shouldReset) {
```

Comment:
NIT: Passing the negated boolean here seems over the top, it'd be better to if check `isEnabled` during the invocation.

Reply:
Yeah, good call @LukasPaczos